### PR TITLE
feat(spec): add spec for initializing the state from the route

### PR DESCRIPTION
**Pull request by yannickcr**
Add a spec for initializing the state from the route.

Will test the following scenario:

1. It navigates to the e-commerce demo with refinements set in route
2. It must have "mixer" set as initial search box value
3. It must have "Appliances" and "Small Kitchen Appliances" items selected in the category menu
4. It must have "KitchenAid" brand selected in the brands menu
5. It must have lower price set to $50 and the upper price set to $350 in the price range
6. It must have free shipping box checked
7. It must have rating "4 & up" selected in the rating menu
8. It must have "Price descending" selected in the sort select
9. It must have "32 hits per page" selected in the hits per page select
10. It must have page 2 selected in the pagination
11. It must have the expected results
12. It sets "cooktop" as search box value
13. It deselects "KitchenAid" brand in the brands menu
14. It deselects "Small Kitchen Appliances" items and select "Ranges, Cooktops & Ovens" in the category menu
15. It selects "Whirlpool" brand in the brands menu
16. It unchecks free shipping box
17. It selects rating "3 & up" in the rating menu
18. It selects "Price ascending" in the sort select
19. It sets lower price to $250 and the upper price to $1250 in the price range
20. It selects "64 hits per page" in the hits per page select
21. It selects page 2 in the pagination
22. It must have the expected url

### CODE REVIEW

File: specs/initial-state-from-route.spec.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
+describe('InstantSearch - Initialize state from the route', () => {
+  it('navigates to the e-commerce demo with refinements set in route', async () => {
+    await browser.url(
+      'examples/e-commerce/search/Appliances%2FSmall+Kitchen+Appliances/?query=mixer&page=2&brands=KitchenAid&rating=4&price=50%3A350&free_shipping=true&sortBy=instant_search_price_desc&hitsPerPage=32'
```
[the remark this answers is not in the document]
Comment (Haroenv) [reply specs/initial-state-from-route.spec.ts]: edit: using QS here will be better

**Comment by yannickcr** [comment]
What do you mean by writing in the router? Making changes in the refinements and check if they are correctly passed in the URL?

**Comment by Haroenv** [comment]
That's how I interpret it as well @yannickcr, making some changes, and checking if the URL is correct

**Comment by yannickcr** [comment]
Ok, in that case I think it can be added to this test.

**Comment by francoischalifour** [comment]
We only test one part of the routing here. We're checking that:

- [x] The URL is translated into the correct UI state
- [ ] The UI state is translated into the correct URL

File: specs/initial-state-from-route.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (yannickcr) [reply specs/initial-state-from-route.spec.ts]: I used [URLSearchParams](https://nodejs.org/docs/latest-v8.x/api/url.html#url_class_urlsearchparams) to get something equivalent to `qs` without any extra dependency https://github.com/algolia/instantsearch-e2e-tests/pull/4/commits/e6c479e0ea737f66836edc328c8d72d0abaebc2c

File: specs/initial-state-from-route.spec.ts [hunk 2]
```diff
@@ -1,7 +1,19 @@
+import { URLSearchParams } from 'url';
+
 describe('InstantSearch - Initialize state from the route', () => {
   it('navigates to the e-commerce demo with refinements set in route', async () => {
+    const params = new URLSearchParams({
```
Comment (Haroenv) [inline specs/initial-state-from-route.spec.ts]: good solution!

**Pull request merged**